scripts/Dockerfile.cardinal-conda-build

```diff
@@ -0,0 +1,54 @@
+# Build and install Cardinal using the MOOSE conda environment in a Docker image
+#
+# Divided into the following targets:
+#
+# cardinal-conda-base: starting from an Ubuntu 24.04 image:
+#     * installs minimal packages
+#     * downloads and runs the miniforge installer
+#     * runs conda init and adds the INL channel
+#     * creats a moose environment
+#
+# cardinal-conda-deps:
+#     * clone cardinal repo
+#     * run the get-dependency script
+#
+# cardinal:
+#     * make cardinal using specific environment variables
+#
+FROM ubuntu:24.04 AS cardinal-conda-base
+
+RUN apt update --yes
+RUN apt install --yes curl cmake mpich git
+
+WORKDIR /cardinal_build
+
+RUN curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh && \
+    bash Miniforge3-Linux-x86_64.sh -b -p ~/miniforge && \
+    rm Miniforge3-Linux-x86_64.sh
+
+ENV PATH=/cardinal_build/miniforge/bin:$PATH
+
+RUN echo $PATH
+RUN conda init --all 
+
+RUN conda update --all --yes && \
+    conda config --add channels https://conda.software.inl.gov/public && \
```

Comment:
I assume INL has plans to create a conda package for Cardinal, as I noticed a repo is already available. I wonder when it will be released.

Reply:
@loganharbour can maybe answer this?

Reply:
This file can be removed. It's actually not used.

We release every master commit of cardinal as a built container in https://hub.docker.com/r/idaholab/cardinal/tags.

Reply:
To be clear, this file is being considered as a contribution - it doesn't exist yet and thus there is nothing to remove...

I made this PR because some folks may want to build their own docker images for development purposes, and this provides a shortcut to figuring out those steps.  Do you share the Dockerfile used to build the images shared on Docker Hub?